central/graphql/resolvers/images_test.go

```diff
@@ -0,0 +1,268 @@
+				return strings.HasPrefix(v.GetCve(), "cve-2019-2")
+			},
+		},
+		{
+			desc:            "filter by severity",
+			q:               PaginatedQuery{Query: pointers.String("Severity:CRITICAL_VULNERABILITY_SEVERITY")},
+			deploymentFiler: func(d *storage.Deployment) bool { return true },
+			imageFilter: func(img *storage.Image) bool {
+				for _, component := range img.GetScan().GetComponents() {
+					for _, vuln := range component.GetVulns() {
+						if strings.HasPrefix(vuln.GetSeverity().String(), "CRITICAL_VULNERABILITY_SEVERITY") {
+							return true
+						}
+					}
+				}
+				return false
+			},
+			vulnFilter: func(v *storage.EmbeddedVulnerability) bool {
+				return strings.HasPrefix(v.GetSeverity().String(), "CRITICAL_VULNERABILITY_SEVERITY")
+			},
+		},
+		{
+			desc:            "filter by severity+fixable",
+			q:               PaginatedQuery{Query: pointers.String("Severity:UNSET_VULNERABILITY_SEVERITY+Fixable:true")},
+			deploymentFiler: func(d *storage.Deployment) bool { return true },
+			imageFilter: func(img *storage.Image) bool {
+				for _, component := range img.GetScan().GetComponents() {
+					for _, vuln := range component.GetVulns() {
+						if strings.HasPrefix(vuln.GetSeverity().String(), "UNSET_VULNERABILITY_SEVERITY") &&
+							vuln.GetFixedBy() != "" {
+							return true
+						}
+					}
+				}
+				return false
+			},
+			vulnFilter: func(v *storage.EmbeddedVulnerability) bool {
+				return strings.HasPrefix(v.GetSeverity().String(), "CRITICAL_VULNERABILITY_SEVERITY") && v.GetFixedBy() != ""
+			},
+		},
+	} {
+		s.T().Run(tc.desc, func(t *testing.T) {
+			paginatedQ := tc.q
+
+			// Test ImageCount query.
+			expectedImages, expectedDeploymentsPerImage := compileExpectedForImageGraphQL(s.testDeployments, s.testImages, tc.deploymentFiler, tc.imageFilter)
+			count, err := s.resolver.ImageCount(ctx, RawQuery{Query: paginatedQ.Query})
+			assert.NoError(t, err)
+			assert.Equal(t, int32(len(expectedImages)), count)
+
+			// Test Images query.
+			actualImages, err := s.resolver.Images(ctx, paginatedQ)
+			assert.NoError(t, err)
+			var expectedIDs []string
+			for _, dep := range expectedImages {
+				expectedIDs = append(expectedIDs, dep.GetId())
+			}
+			assert.ElementsMatch(t, expectedIDs, getIDList(ctx, actualImages))
+
+			for _, image := range actualImages {
+				imageID := string(image.Id(ctx))
+				// Test DeploymentCount field for each image resolver.
+				expectedDeployments := expectedDeploymentsPerImage[imageID]
+				actualDeploymentCount, err := image.DeploymentCount(ctx, RawQuery{Query: paginatedQ.Query})
+				assert.NoError(t, err)
+				assert.Equal(t, int32(len(expectedDeployments)), actualDeploymentCount)
+
+				if !features.VulnMgmtWorkloadCVEs.Enabled() {
+					return
+				}
+
+				// Test ImageCVECountBySeverity for each deployment resolver.
+				expectedCVESevCount := compileExpectedCountBySeverity([]*storage.Image{expectedImages[imageID]}, tc.vulnFilter)
```

Comment:
Move `compileExpectedCountBySeverity` from resolvers/deployments_test.go to resolvers/test_utils.go